templates/CRM/common/customDataBlock.tpl

```diff
@@ -0,0 +1,17 @@
+{if $customDataType}
```

Comment:
@eileenmcnaughton do these variables get automatically passed through to the new template?

Reply:
The variables are assigned from php & are available for all smarty templates - but I think you are referring to the addition of an outer-if. If you look lower - that is a required param - see 

CRM.buildCustomData( '{$customDataType}' );

So the if means ' don't include if not actually valid' - which would not affect this case but makes it 'includable' across a broader range of entities - ie. ones where it would not be valid unless an extension has altered cg_extends to broaden the entities definable

Reply:
I have been having problems to add custom fields with this structure since the buildCustomData() function adds the custom groups html into div#customData.

Consider an example if you want to add a tab 'Gifts' on contribution page settings screen which is an entity and want to enable ability to use custom field for Gift entity. When i use entity form structure  it works when i open as standalone form but when i try to embed into the tab it adds on the 'Title and Settings'  form unless you do some hackish way to add into required form.